geoh5py/ui_json/parameters.py

```diff
@@ -0,0 +1,568 @@
+        self._tooltip: str | None = None
+        self._extra_members: dict[str, Any] = {}
+        self._active_members: list[str] = list(kwargs)
+        self.validations: Validations = Validations(validations)
+        self.register(kwargs)
+
+    @classmethod
+    def from_dict(
+        cls, name: str, form: dict[str, Any], validations: dict | None = None
+    ):
+        return cls(name, validations, **form)
+
+    @property
+    def value(self):
+        return self._value.value
+
+    @value.setter
+    def value(self, val):
+        self._value.value = val
+
+    def register(self, members: dict[str, Any]):
+        """
+        Set parameters from form members with default or incoming values.
+
+        :param members: Dictionary of form members and associated data.
+
+        :return: Dictionary of unrecognized members and data.
+        """
+
+        if not members:
+            return
+
+        if not isinstance(members, dict):
+            raise TypeError("Input 'members' must be a dictionary.")
+
+        members = {KEY_MAP.get(k, k): v for k, v in members.items()}
+        error_list = []
+        for member in self.valid_members:
+            validations = (
+                self.validations if member == "value" else self.form_validations[member]
+            )
+            if member in members:
+                try:
+                    param = Parameter(
+                        member, members.pop(member), validations  # type: ignore
+                    )
+                except BaseValidationError as err:
+                    error_list.append(err)
+            else:
+                param = Parameter(member, validations=validations)  # type: ignore
+
+            setattr(self, f"_{member}", param)
+            if member not in dir(self):  # do not override pre-defined properties
+                setattr(self.__class__, member, ValueAccess(f"_{member}"))
+
+        if error_list:
+            if len(error_list) == 1:
+                raise error_list.pop()
+
+            raise AggregateValidationError(self.name, error_list)
+
+        self._extra_members.update(members)
+
+    @property
+    def validations(self) -> Validations:
+        return self._value.validations
+
+    @validations.setter
+    def validations(self, val):
+        self._value.validations = val
+
+    @property
+    def active(self) -> list[str]:
+        """
+        Returns names of active form members.
+
+        :return: List of active form members.  These will include any members
+            that were:
+                1. Provided during construction.
+                2. Updated through the 'register' method.
+                3. Updated through member setters.
+                4. Defined as 'required' by the validations.
+        """
+        active = self._active_members + list(self._extra_members)
+        active_unique, ind = np.unique(active, return_index=True)
+        return list(active_unique[ind])  # Preserve order after unique

```

Comment:
Consider using `list(set(self._active_members + list(self._extra_members)))` instead of numpy unique.

Reply:
I think it's better if we preserve order.  Both np.unique (without the re-index) and the set sort the lists alphabetically.